R/TileDBArray.R

```diff
@@ -1498,6 +1498,14 @@ setMethod(
             ase <- tiledb_array_schema_evolution_extend_enumeration(ase, arr, allnames[k], added_enums)
             tiledb::tiledb_array_schema_evolution_array_evolve(ase, uri)
             value[[k]] <- factor(value[[k]], levels = unique(c(dictionary, added_enums)), ordered = is.ordered(value[[k]]))
+
+            } else if (isFALSE(setequal(new_levels, dictionary)) && isFALSE(is.null(new_levels))) {
```

Comment:
Both `setequal()` and `is.null()` return scalar logicals and never return `NA`, so we can safely use `!` instead of checking with `isFALSE()`
```suggestion
            } else if (!setequal(new_levels, dictionary) && !is.null(new_levels)) {
```